### PR TITLE
[upload, download] - fix bugs with lists

### PR DESCRIPTION
**Pull request by GuyAfik**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-27578, https://jira-hq.paloaltonetworks.local/browse/CIAC-8354

## Description

on this situation downloading lists would be skipped
image

1. it is not true that in order to identify list we should use `allRead` key, is is not always part of the `List` structure.
2. `data` key on the other hand **always appear at any list in any variation** and also the `truncated` key.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 ## Unreleased
 * Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with brackets that contains a dot at the end of them.
+* Fixed an issue where the **upload** failed when trying to upload a `List`.
+* Fixed an issue where the **download** would skip downloading `Lists` that are assigned to specific user roles with no roles.
```
Comment (dantavori) [inline CHANGELOG.md:5]:
```suggestion
* Fixed an issue where the **upload** failed when trying to upload a list content item.
* Fixed an issue where the **download** would skip downloading list content items assigned to specific user roles with no roles.
```

File: demisto_sdk/commands/content_graph/objects/list.py [hunk 2]
```diff
@@ -19,13 +19,10 @@ def _upload(
         client: demisto_client,
         marketplace: MarketplaceVersions,
     ) -> None:
-        with TemporaryDirectory("w") as f:
-            dir_path = Path(f)
-            self.dump(dir_path, marketplace=marketplace)
-
+        with self.path.open("r") as _list:
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/objects/list.py:22]: why is this change? (@dorschw fyi)

Comment (GuyAfik) [reply demisto_sdk/commands/content_graph/objects/list.py:22]: fixes https://jira-hq.paloaltonetworks.local/browse/CIAC-8354
basically we are trying to send a `Path` object in the api request, leading to the following mentioned in the issue.

uploading a list is broken ATM, this PR fixes it

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 ## Unreleased
 * Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with brackets that contains a dot at the end of them.
 * Fixed an issue where running **validate -g** failed reading a `.pack-ignore` file that contained only newlines and spaces.
+* Fixed an issue where the **upload** failed when trying to upload a list content item.
+* Fixed an issue where the **download** would skip downloading list content items assigned to specific user roles with no roles.
```
Comment (dorschw) [inline CHANGELOG.md:6]:
```suggestion
* Fixed an issue where **upload** failed when trying to upload a list-type content item.
* Fixed an issue where **download** would skip list-type items assigned to specific user roles with no roles.
```

image

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -1906,6 +1903,9 @@ def find_type(
         if "rule_id" in _dict:
             return FileType.LAYOUT_RULE
 
+        if "data" in _dict or "allRead" in _dict or "truncated" in _dict:
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:1906]:
```suggestion
        if {"data", "allRead", "truncated"}.intersection(_dict.keys()):
```

**Review by dantavori — APPROVED** [review]

File: demisto_sdk/commands/content_graph/objects/list.py [hunk 5]
```diff
@@ -23,9 +25,10 @@ def _upload(
...
-                method="POST",
-                path="lists/save",
-                body=dir_path / self.normalize_name,
-                response_type="object",
-            )
+            with (dir_path / self.normalize_name).open("r") as _list:
+                client.generic_request(
+                    method="POST",
+                    path="lists/save",
+                    body=json.load(_list),
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/list.py:32]:
```suggestion
                    body=json.loads((dir_path / self.normalize_name).read_text()),
```

File: demisto_sdk/commands/content_graph/objects/list.py [hunk 6]
```diff
@@ -23,9 +25,10 @@ def _upload(
             dir_path = Path(f)
             self.dump(dir_path, marketplace=marketplace)
 
-            client.generic_request(
-                method="POST",
-                path="lists/save",
-                body=dir_path / self.normalize_name,
-                response_type="object",
-            )
+            with (dir_path / self.normalize_name).open("r") as _list:
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/list.py:28]: remove

**Pull request merged**